worker/net.go

```diff
@@ -3,9 +3,13 @@ package worker
 import (
 	"context"
 	"net"
+
+	"go.mercari.io/go-dnscache"
+	"go.sia.tech/renterd/internal/utils"
 )
 
 func dial(ctx context.Context, hostIP string) (net.Conn, error) {
-	conn, err := (&net.Dialer{}).DialContext(ctx, "tcp", hostIP)
+	dialFunc := dnscache.DialFunc(utils.Resolver, (&net.Dialer{}).DialContext)
```

Comment:
My preference would be to pass in an optional dial func as a field on the worker rather than hardcoding it to this resolver.